irmaseal-core/src/meta.rs

```diff
@@ -0,0 +1,95 @@
+use super::stream::*;
+use crate::Identity;
+use super::{Error};
+use base64;
+use arrayvec::{ArrayString, ArrayVec};
+use serde::{Deserialize, Serialize, Deserializer, Serializer};
+
+/// The version of the encrypted data
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum Version {
+    V1_0,
+}
+
+/// Metadata which contains the version
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct Metadata {
+    pub version: Version,
+    pub media_type: ArrayString<[u8; 255]>,
+    pub media_metadata: serde_json::Value,
+    #[serde(serialize_with = "to_base64_ciphertext", deserialize_with = "from_base64_ciphertext")]
+    pub ciphertext: ArrayVec<[u8; CIPHERTEXT_SIZE]>,
+    #[serde(serialize_with = "to_base64_iv", deserialize_with = "from_base64_iv")]
+    pub iv: ArrayVec<[u8; IVSIZE]>,
```

Comment:
`iv` has a static size of exact 16 bytes and `ciphertext` of exact 144 bytes. This means that an ArrayVec is not necessary for this, it can just be an array with a static length. Can serde handle this?

Reply:
What's the difference between an ArrayVec and a normal array? No where in the code base normal arrays are used so I avoided it as well.

Reply:
ArrayVec is an array wrapped in a package to make it possible to let it contain less than `size` values. It grows like a vector with the only difference that an array is used under water to store the values, so if you put more items in than the size of the underlying array you'll get an error.

In this case you don't need the growing functionality, since you exactly know how many items there will be. Therefore you can just say `iv: [u8; IVSIZE]`. The only question is then, can Serde parse the iv directly into a slice/array? Otherwise it is of course not possible to use a fixed array here.

Reply:
`iv: [u8; IVSIZE]` doesn't work because rust array only work upto a specific length due to const generics not working yet,